version: "3.9"

volumes:
  postgres_database_data:

services:
  postgres:
    image: postgres:latest
    env_file:
      - ./.env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
    # Saving data on local
    volumes:
      - "postgres_database_data:/var/lib/postgresql/data/"

  django:
    image: oil-and-rope:0.1
    build:
      context: .
      dockerfile: ./compose/django/Dockerfile
    depends_on:
      postgres:
        condition: service_healthy
    environment:
      DB_HOST: postgres
    env_file:
      - ./.env
    volumes:
      - "./compose/django/command.sh:/command.sh"
      - "./:/app/"
    command: /command.sh
    working_dir: /app/

  nginx:
    image: nginx:latest
    depends_on:
      - django
    ports:
      - "127.0.0.1:8000:80/tcp"
      - "127.0.0.1:8443:433/tcp"
    volumes:
      - "./static:/static"
      - "./media:/media"
      - "./compose/nginx/conf.d/:/etc/nginx/conf.d/"
      - "./compose/nginx/nginx.conf:/etc/nginx/nginx.conf"